### PR TITLE
Feat/cleanup part 2

### PR DESCRIPTION
**Pull request by ditchfieldcaleb**
Convert remaining helper utilities & scripts to typescript.
Add proper ethers-v5 contract types.

### CODE REVIEW

File: scripts/utils/exec.ts [hunk 1]
```diff
@@ -13,28 +15,22 @@ async function exec(txData, totalValue, addresses) {
...
+    ).sendTransaction({
       to: governorAlphaAddress,
       data: txData,
-      gas: 6000000
+      gasLimit: 6000000
     });
   }
 
-  const governor = await GovernorAlpha.at(governorAlphaAddress);
+  const governor = await ethers.getContractAt('GovernorAlpha.', governorAlphaAddress);
```
Comment (Joeysantoro) [inline scripts/utils/exec.ts:33]: is there supposed to be a period in here?

File: scripts/utils/exec.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
-const hre = require('hardhat');
+import hre, { ethers, artifacts } from 'hardhat';
+import { time } from '@openzeppelin/test-helpers';
 
-const GovernorAlpha = artifacts.require('GovernorAlpha');
+import * as dotenv from 'dotenv';
+
+dotenv.config();
+
+const GovernorAlpha = artifacts.readArtifactSync('GovernorAlpha');
```
Comment (Joeysantoro) [inline scripts/utils/exec.ts:8]: is this used?

File: scripts/utils/helpers.ts [hunk 3]
```diff
@@ -10,12 +13,10 @@ function check(flag, message) {
 
 // The current version of the contracts in the repo uses readOracle as the api
 // The old api was peg(), so the currently deployed contracts need manual calling until the upgrade
-async function readOracle(oracleRef, web3) {
-  const data = await web3.eth.call({ to: oracleRef.address, data: web3.eth.abi.encodeFunctionSignature('peg()') });
+export async function readOracle(oracleRef, web3) {
+  const data = await ethers.provider.call({
+    to: oracleRef.address,
+    data: web3.eth.abi.encodeFunctionSignature('peg()')
```
Comment (Joeysantoro) [inline scripts/utils/helpers.ts:19]: seeing web3 and ethers together is starting to mess with me. How hard will it be to axe web3 completely?

File: types/types.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
+import { ethers } from 'ethers';
```
Comment (Joeysantoro) [inline types/types.ts:1]: For my knowledge can you explain again the benefits of adding this file. Is this what gives us granular types per contract?

File: scripts/utils/syncPool.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+// the pair would sync to that price
+export async function syncPool(targetBPs: number, addresses: NamedAddresses, sendingAddress: string, logging = false) {
+  const feiAddress = addresses.feiAddress;
+  const ethUniswapPCVDepositAddress = addresses.ethUniswapPCVDepositAddress;
+  const ethPairAddress = addresses.ethPairAddress;
+  const fei = await ethers.getContractAt('Fei', feiAddress);
+  const uniswapPcvDeposit = await ethers.getContractAt('UniswapPCVDeposit', ethUniswapPCVDepositAddress);
+  const ethPair = await ethers.getContractAt('IUniswapV2Pair', ethPairAddress);
+
+  logging ? console.log('Current') : undefined;
```
Comment (Joeysantoro) [inline scripts/utils/syncPool.ts:20]: nit: we started doing logging && console.log() to remove the `: undefined` if you want to edit

Comment (ditchfieldcaleb) [reply scripts/utils/syncPool.ts:20]: Will do!

File: scripts/utils/exec.ts (hunk 2 above)
Comment (sol-sr) [reply scripts/utils/exec.ts:8]: We should have a linter that runs locally and in CI that catches unused vars.

File: scripts/utils/exec.ts (hunk 1 above)
Comment (ditchfieldcaleb) [reply scripts/utils/exec.ts:33]: No - fixed.

File: scripts/utils/exec.ts (hunk 2 above)
Comment (ditchfieldcaleb) [reply scripts/utils/exec.ts:8]: Nope, removed.

File: scripts/utils/helpers.ts (hunk 3 above)
Comment (ditchfieldcaleb) [reply scripts/utils/helpers.ts:19]: SOON

File: scripts/utils/exec.ts (hunk 2 above)
Comment (ditchfieldcaleb) [reply scripts/utils/exec.ts]: @sol-sr absolutely we should. That's in the pipeline

**Pull request merged**